src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs

```diff
@@ -35,7 +35,11 @@ internal void SetOwner(MediaPlayerElement owner)
 			}
 			else
 			{
-				return (float)naturalOtherDimension * scaledOneDimension / (float)naturalOneDimension;
+				// There are situations where between measurements scaledDimension and naturalDimension
+				// have a small difference in value (a few pixels) causing the measure to go into an infinite loop.
+				// Related to: https://github.com/unoplatform/uno/issues/15254
+				var ratio = (float)Math.Round(scaledOneDimension / naturalOneDimension, 0);
```

Comment:
Should we make this rounding happen only on iOS?

Reply:
@Youssef1313  I thought about it and it shouldn't be a problem making it iOS-Only. Another thing I thought was keeping it this way as it would not affect much or anything other platforms.

@jeromelaban : what do you think?

Reply:
I went ahead and pushed a change to do the rounding only on iOS. This will limit the impact.